aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/DeleteHandler.java

```diff
@@ -50,4 +56,22 @@
             throw new CfnInvalidRequestException(ResourceModel.TYPE_NAME + e.getMessage(), e);
         }
     }
+
+    private void deleteNotificationChannels(final String pgName, final AmazonWebServicesClientProxy proxy) {
+        GetNotificationConfigurationRequest getNotificationConfigurationRequest = GetNotificationConfigurationRequest.builder()
+                .profilingGroupName(pgName)
+                .build();
+
+        GetNotificationConfigurationResponse getNotificationConfigurationResponse = proxy.injectCredentialsAndInvokeV2(getNotificationConfigurationRequest,
+                profilerClient::getNotificationConfiguration);
+
+        // Iterate through all channels and remove them
+        getNotificationConfigurationResponse.notificationConfiguration().channels().forEach(channel -> {
+            RemoveNotificationChannelRequest removeNotificationChannelRequest = RemoveNotificationChannelRequest.builder()
+                    .profilingGroupName(pgName)
+                    .channelId(channel.id())
+                    .build();
+            proxy.injectCredentialsAndInvokeV2(removeNotificationChannelRequest, profilerClient::removeNotificationChannel);
+        });
```

Comment:
minor/thought: Couldn't this be done on the `DeleteProfilingGroup` API implicitly?